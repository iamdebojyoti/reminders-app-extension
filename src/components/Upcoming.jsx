import PropTypes from "prop-types";
import { Accordion, Card, Button } from "react-bootstrap";
import { CheckCircle } from "./Icons";
import AccordionToggle from "./AccordionToggle";

const Upcoming = ({reminders, complete}) => {

  const backgroundVariant = (idx) => {
    if(idx % 2 == 0) {
      return "light"
    } else {
      return "dark"
    }
  };

  const textVariant = (idx) => {
    if(idx % 2 == 0) {
      return "dark"
    } else {
      return "light"
    }
  };

  return (
    <>
      {reminders.map((reminder, idx) => {
        return (
          <Accordion defaultActiveKey="0" key={idx}>
            <Card
              bg={backgroundVariant(idx)}
              key={idx}
              text={textVariant(idx)}
              className="mb-2"
            >
              <Card.Header>
                {reminder.name}
                <Button
                  id={"complete-"+reminder.id}
                  className="float-end"
                  variant={backgroundVariant(idx)}
                  onClick={() => complete(reminder.id)}
                >
                  <CheckCircle />
                </Button>
                <AccordionToggle eventKey={idx} variant={backgroundVariant(idx)}/>
              </Card.Header>
              <Accordion.Collapse eventKey={idx}>
                <Card.Body>
                  <Card.Text>{reminder.description}</Card.Text>
                </Card.Body>
              </Accordion.Collapse>
            </Card>
          </Accordion>
        );
      })}
    </>
  );
};

Upcoming.propTypes = {
  complete: PropTypes.func.isRequired,
  reminders: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      state: PropTypes.string.isRequired,
    })
  ),
};

export default Upcoming;
