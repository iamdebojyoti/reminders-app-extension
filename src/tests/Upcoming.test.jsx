import { shallow } from "enzyme";
import Upcoming from "../components/Upcoming";

describe('Testing <Upcoming/> component', () => {

    let component, props;

    beforeEach(() => {
        props = {
            complete: jest.fn(),
            reminders: [{
                id: '1',
                name: 'Reminder-1',
                description: 'Reminder-desc-1',
                state: 'created'
            }, {
                id: '2',
                name: 'Reminder-2',
                description: 'Reminder-desc-2',
                state: 'created'
            }, {
                id: '3',
                name: 'Reminder-3',
                description: 'Reminder-desc-3',
                state: 'created'
            }],
        };
    });

    it('should match snapshot of Upcoming at initial state', () => {
        component = shallow(<Upcoming {...props}/>);
        expect(component.debug()).toMatchSnapshot();
    });

    it('should call done when complete button is clicked', () => {
        component = shallow(<Upcoming {...props}/>);
        component.find('#complete-2').simulate('click');
        component.update();
        expect(props.complete).toHaveBeenCalledWith("2");
    })
});