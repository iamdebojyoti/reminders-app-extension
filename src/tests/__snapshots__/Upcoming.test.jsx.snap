// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Upcoming/> component should match snapshot of Upcoming at initial state 1`] = `
"<Fragment>
  <Accordion defaultActiveKey=\\"0\\">
    <Card bg=\\"light\\" text=\\"dark\\" className=\\"mb-2\\" body={false}>
      <CardHeader>
        Reminder-1
        <Button id=\\"complete-1\\" className=\\"float-end\\" variant=\\"light\\" onClick={[Function: onClick]} active={false} disabled={false}>
          <CheckCircle />
        </Button>
        <AccordionToggle eventKey={0} variant=\\"light\\" />
      </CardHeader>
      <AccordionCollapse eventKey={0}>
        <CardBody>
          <CardText>
            Reminder-desc-1
          </CardText>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </Accordion>
  <Accordion defaultActiveKey=\\"0\\">
    <Card bg=\\"dark\\" text=\\"light\\" className=\\"mb-2\\" body={false}>
      <CardHeader>
        Reminder-2
        <Button id=\\"complete-2\\" className=\\"float-end\\" variant=\\"dark\\" onClick={[Function: onClick]} active={false} disabled={false}>
          <CheckCircle />
        </Button>
        <AccordionToggle eventKey={1} variant=\\"dark\\" />
      </CardHeader>
      <AccordionCollapse eventKey={1}>
        <CardBody>
          <CardText>
            Reminder-desc-2
          </CardText>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </Accordion>
  <Accordion defaultActiveKey=\\"0\\">
    <Card bg=\\"light\\" text=\\"dark\\" className=\\"mb-2\\" body={false}>
      <CardHeader>
        Reminder-3
        <Button id=\\"complete-3\\" className=\\"float-end\\" variant=\\"light\\" onClick={[Function: onClick]} active={false} disabled={false}>
          <CheckCircle />
        </Button>
        <AccordionToggle eventKey={2} variant=\\"light\\" />
      </CardHeader>
      <AccordionCollapse eventKey={2}>
        <CardBody>
          <CardText>
            Reminder-desc-3
          </CardText>
        </CardBody>
      </AccordionCollapse>
    </Card>
  </Accordion>
</Fragment>"
`;
